system/jlib/jptree.cpp

```diff
@@ -8643,19 +8644,113 @@ MODULE_EXIT()
+        virtual void setPropBin(const char *xpath, size32_t size, const void *data) override {}
+        virtual void addPropBin(const char *xpath, size32_t size, const void *data) override {}
+        virtual void appendPropBin(const char *xpath, size32_t size, const void *data) override {}
+
+        virtual IPropertyTree *getPropTree(const char *xpath) const override { return nullptr; }
+        virtual IPropertyTree *queryPropTree(const char *xpath) const override { return nullptr; }
+        virtual IPropertyTree *setPropTree(const char *xpath, IPropertyTree *val) override { return nullptr; }
+        virtual IPropertyTree *addPropTree(const char *xpath, IPropertyTree *val) override { return nullptr; }
+
+        virtual IPropertyTree *setPropTree(const char *xpath) override { return nullptr; }
+        virtual IPropertyTree *addPropTree(const char *xpath) override { return nullptr; }
+
+        virtual bool removeProp(const char *xpath) override { return false; }
+        virtual bool removeTree(IPropertyTree *child) override { return false; }
+        virtual aindex_t queryChildIndex(IPropertyTree *child) override { return 0; }
+
+        virtual StringBuffer &getName(StringBuffer &ret) const override { return ret; }
+        virtual const char *queryName() const override { return nullptr; }
+
+        virtual IPropertyTreeIterator *getElements(const char *xpath, IPTIteratorCodes flags = iptiter_null) const override { return nullptr; }
+        virtual IAttributeIterator *getAttributes(bool sorted=false) const override { return nullptr; }
+
+        virtual IPropertyTree *getBranch(const char *xpath) const override { return nullptr; }
+        virtual IPropertyTree *queryBranch(const char *xpath) const override { return nullptr; }
+        virtual bool hasChildren() const override { return false; }
+        virtual unsigned numUniq() const override { return 0; }
+        virtual unsigned numChildren() const override { return 0; }
+        virtual bool isCaseInsensitive() const override { return false; }
+        virtual bool IsShared() const override { return false; }
+        virtual void localizeElements(const char *xpath, bool allTail=false) override {}
+        virtual unsigned getCount(const char *xpath) const override { return 0; }
+        virtual IPropertyTree *addPropTreeArrayItem(const char *xpath, IPropertyTree *val) override { return nullptr; }
+        virtual bool isArray(const char *xpath=NULL) const override { return false; }
+        virtual unsigned getAttributeCount() const override { return 0; }
+    };
+    return new NullIPropertyTree;
+}
+
 IPropertyTree * getComponentConfig()
 {
     CriticalBlock b(configCS);
     if (!componentConfiguration)
-        throw makeStringException(99, "Configuration file has not yet been processed");
+    {
+        if (!nullConfiguration)
+        {
+            PrintStackReport(); // deliberately emitting here, so only reported once per process
```

Comment:
Do we want to do this for every component that used to explicitly init a null config ?

Reply:
I agree, would be better if did not.
I am going to reintroduce initNullConfiguration(), meaning that the normal case for components that have no config, is that they should call this (as they did), but now (other changes in this PR) will mean any code elsewhere that tries to use the config when either there isn't one, or it's called too early (e.g. as in module static code like in HPCC-33418) will not cause failure - but will instead warn and print a stack trace.

Reply:
@mckellyln - please see 2nd commit.